internal/dataplane/parser/translators/ingress_atc_test.go

```diff
@@ -0,0 +1,259 @@
+					Host:           kong.String("test-service.default.80.svc"),
+					ConnectTimeout: kong.Int(int(defaultServiceTimeout.Milliseconds())),
+					Path:           kong.String("/"),
+					Port:           kong.Int(80),
+					Protocol:       kong.String("http"),
+					Retries:        kong.Int(defaultRetries),
+					ReadTimeout:    kong.Int(int(defaultServiceTimeout.Milliseconds())),
+					WriteTimeout:   kong.Int(int(defaultServiceTimeout.Milliseconds())),
+				},
+				Routes: []kongstate.Route{{
+					Ingress: util.K8sObjectInfo{
+						Name:      "test-ingress",
+						Namespace: corev1.NamespaceDefault,
+					},
+					Route: kong.Route{
+						Name:              kong.String("default.test-ingress.test-service.konghq.com.80"),
+						Expression:        kong.String(`(http.host == "konghq.com") && (http.path ^= "/api/") && ((net.protocol == "http") || (net.protocol == "https"))`),
+						Priority:          kong.Int(NormalIngressExpressionPriority),
+						PreserveHost:      kong.Bool(true),
+						StripPath:         kong.Bool(false),
+						ResponseBuffering: kong.Bool(true),
+						RequestBuffering:  kong.Bool(true),
+						Tags:              kong.StringSlice("k8s-name:test-ingress", "k8s-namespace:default"),
+					},
+					ExpressionRoutes: true,
+				}},
+				Backends: []kongstate.ServiceBackend{{
+					Name:      "test-service",
+					Namespace: corev1.NamespaceDefault,
+					PortDef: kongstate.PortDef{
+						Mode:   kongstate.PortModeByNumber,
+						Number: 80,
+					},
+				}},
+				Parent: expectedParentIngress(),
+			}},
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			services := TranslateIngress(tc.ingress, false, true)
+			checkOnlyObjectMeta := cmp.Transformer("checkOnlyObjectMeta", func(i *netv1.Ingress) *netv1.Ingress {
+				// In the result we only care about ingresses' metadata being equal.
+				// We ignore specification to simplify tests.
+				return &netv1.Ingress{
+					ObjectMeta: i.ObjectMeta,
+				}
+			})
+			diff := cmp.Diff(tc.expectedServices, services, checkOnlyObjectMeta)
+			require.Empty(t, diff, "expected no difference between expected and translated ingress")
+		})
+	}
+}
+
+func TestPathMatcherFromIngressPath(t *testing.T) {
+	testCases := []struct {
+		name        string
+		path        netv1.HTTPIngressPath
+		regexPrefix string
+		expression  string
+	}{
+		{
+			name: "simple prefix match",
+			path: netv1.HTTPIngressPath{
+				Path:     "/v1/api",
+				PathType: &pathTypePrefix,
+			},
+			expression: `(http.path == "/v1/api") || (http.path ^= "/v1/api/")`,
+		},
+		{
+			name: "simple exact match",
+			path: netv1.HTTPIngressPath{
+				Path:     "/v1/api",
+				PathType: &pathTypeExact,
+			},
+			expression: `http.path == "/v1/api"`,
+		},
+		{
+			name: "regex match",
+			path: netv1.HTTPIngressPath{
+				Path:     "/~/[a-z]+",
+				PathType: &pathTypeImplementationSpecific,
+			},
+			expression: `http.path ~ "^/[a-z]+"`,
+		},
+		{
+			name: "empty path",
```

Comment:
```suggestion
			name: "empty prefix path",
```
I think that we could also add two more cases for empty paths for regex and exact types.

Reply:
added more tests on paths.